public/js/extended-access.js

```diff
@@ -0,0 +1,240 @@
+
+      insertHighlightedJson("#output", parseJwt(gaaUser.credential), "GAA User Credentials from the JWT")
+
+      createCookie('jwtSub', parseJwt(gaaUser.credential).sub)
+      // gaaUser - registration JWT that is returned by Sign In with Google
+      // https://developers.google.com/identity/gsi/web/reference/js-reference?hl=en#credential
+      const gaaUserDecoded = parseJwt(gaaUser.credential);
+      // Example body of the promise returning a userState object
+      const userState = {
+        id: btoa(gaaUserDecoded.sub),
+        registrationTimestamp: Math.round(Date.now() / 1000),
+        granted: false
+      };
+      resolve(userState);
+    });
+  });
+
+  // Allows the user to login to an existing publisher account and must resolve
+  // with an updated userState object for the newly logged-in user.
+  // Do not copy the implementation for the promise and see documentation for your own implementation at:
+  // https://developers.google.com/news/subscribe/extended-access/integration-steps/web-implementation#handle-login-for-existing-users
+  const handleLoginPromise = new Promise ((resolve) => {
+    GaaMetering.getLoginPromise().then(() => {
+      alert('You may open your log in page');
+      // Example body of the promise returning a userState object
+      const userState = {
+        id: 'user123456789',
+        registrationTimestamp: 1647311022,
+        granted: false,
+      };
+
+      insertHighlightedJson("#output", {message:"User identified via Publisher's sign in"}, "User with Publisher Login")
+      resolve(userState);
+    });
+  });
+
+  // Get userState by passing the gaaUser object to your registration endpoint server
+  // Resolves with the current user’s entitlements from the publisher
+  // Optional - required only if your userState object doesn't contain the publisherEntitlement already
+  // Do not copy the implementation for the promise and see documentation for your own implementation at:
+  // https://developers.google.com/news/subscribe/extended-access/integration-steps/web-implementation#creating-the-users-entitlement-state
+  const publisherEntitlementPromise = new Promise ((resolve) => {
+    // Example body of the promise returning a publisherEntitlement object
+    const publisherEntitlement = {
+      granted: false,
+    };
+    resolve(publisherEntitlement);
+  });
+
+  // The userState object for the current user.
+  // Do not copy the implementation for the callback and see documentation for your own implementation at:
+  // https://developers.google.com/news/subscribe/extended-access/reference/user-state-object
+  function getUserState() {
+    return {
+      // Commented out since we don't actually use the UserState in this demo of the
+      // registration flow.
+      // id: 'user123456789',
+      // registrationTimestamp: 1647311022,
+      // subscriptionTimestamp: 1647311062,
+      // granted: true,
+      // grantReason: 'SUBSCRIBER'
+    };
+  };
+
+  /* Code */
```

Comment:
Is this needed?

Reply:
/shrug it's how we had it in the other examples, so I kept it. I think it makes sense to have the function mocked to illustrate where you'd generate the function?